subscription-service/src/main/kotlin/com/egm/stellio/subscription/service/NotificationService.kt

```diff
@@ -56,22 +59,30 @@ class NotificationService(
         entityId: URI,
         entity: JsonLdEntity
     ): Mono<Triple<Subscription, Notification, Boolean>> {
+        val mediaType = MediaType.valueOf(subscription.notification.endpoint.accept.accept)
         val notification = Notification(
             subscriptionId = subscription.id,
-            data = buildNotificationData(entity, subscription.notification)
+            data = buildNotificationData(entity, subscription.notification, mediaType)
         )
         val uri = subscription.notification.endpoint.uri.toString()
         logger.info("Notification is about to be sent to $uri for subscription ${subscription.id}")
         if (uri == "urn:embedded:firebase") {
             val fcmDeviceToken = subscription.notification.endpoint.getInfoValue("deviceToken")
             return callFCMSubscriber(entityId, subscription, notification, fcmDeviceToken)
         } else {
-            var request = WebClient.create(uri).post() as WebClient.RequestBodySpec
-            subscription.notification.endpoint.info?.forEach {
-                request = request.header(it.key, it.value)
-            }
-            // FIXME hardcoding a JSON content type for now
-            request = request.contentType(MediaType.APPLICATION_JSON)
+            val request =
+                WebClient.create(uri).post().contentType(mediaType).headers {
+                    if (mediaType == MediaType.APPLICATION_JSON) {
+                        if (subscription.contexts.size > 1) {
+                            val generateCompoundLink = "$hostname/ngsi-ld/v1/subscriptions/${subscription.id}/context"
```

Comment:
you call it `hostname` but it seems you're expecting more than an hostname

Reply:
tip : `stellio.egm.io` is a hostname, not `https://stellio.egm.io`

Reply:
`generateCompoundLink` is not a really good name (what are you "generating" here?)